.github/workflows/test.yml

```diff
@@ -42,10 +45,10 @@ jobs:
           cache-name: cache-compiled-build
         with:
           path: _build
-          key: ${{ runner.os }}-mix-${{ env.cache-name }}-${{ hashFiles('**/mix.lock') }}
+          key: ${{ runner.os }}-mix-${{ env.cache-name }}-elixir-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}
           restore-keys: |
-            ${{ runner.os }}-mix-${{ env.cache-name }}-
-            ${{ runner.os }}-mix-
```

Comment:
I had to make these changes because otherwise I believe it was trying to use the same cached version for both combos (1.12/OTP24, 1.16/OTP26) and throwing an error about a BEAM version mismatch